Doc: The default sorting alg. is stable from 1.9 (#47579)
JuliaLang/julia: doc/src/base/sort.md

```

`MergeSort` is an O(n log n) stable sorting algorithm but is not in-place – it requires a temporary
array of half the size of the input array – and is typically not quite as fast as `QuickSort`.
It is the default algorithm for non-numeric data.

The default sorting algorithm is chosen on the basis that it is stable and fast, or *appear*
to be fast. Usually, `QuickSort` is selected, but `InsertionSort` is preferred for small
data. You can also explicitly specify your preferred algorithm, e.g.
`sort!(v, alg=PartialQuickSort(10:20))`.

The mechanism by which Julia picks default sorting algorithms is implemented via the
`Base.Sort.defalg` function. It allows a particular algorithm to be registered as the
default in all sorting functions for specific arrays. For example, here is the default
method from [`sort.jl`](https://github.com/JuliaLang/julia/blob/master/base/sort.jl):

```julia
defalg(v::AbstractArray) = DEFAULT_STABLE
```

You may change the default behavior for specific element type by:
```julia
defalg(v::AbstractArray{<:Number}) = MergeSort
```

!!! compat "Julia 1.9"
    The default sorting algorithm (returned by `Base.Sort.defalg` function) is

>     stable since Julia 1.9.

Author: Updating to `... is is guaranteed to be stable since Julia 1.9.` that should have no side-effect implications.

  * [`PartialQuickSort(k)`](@ref)
  * [`MergeSort`](@ref)

`InsertionSort` is an O(n²) stable sorting algorithm. It is efficient for very small `n`,
and is used internally by `QuickSort`.

`QuickSort` is a very fast sorting algorithm with an average-case time complexity of
O(n log n). `QuickSort` is stable, i.e., elements considered equal will remain in the same
order. Notice that O(n²) is worst-case complexity, but it gets vanishingly unlikely as the
pivot selection is randomized.

`PartialQuickSort(k::OrdinalRange)` is similar to `QuickSort`, but the output array is only
sorted in the range of `k`. For example:

```jldoctest
julia> x = rand(1:500, 100);

julia> k = 50:100;

julia> s1 = sort(x; alg=QuickSort);

julia> s2 = sort(x; alg=PartialQuickSort(k));

julia> map(issorted, (s1, s2))
(true, false)

julia> map(x->issorted(x[k]), (s1, s2))
(true, true)

julia> s1[k] == s2[k]
true
```

!!! compat "Julia 1.9"

>     The `QuickSort` and `PartialQuickSort` are stable since Julia 1.9.

Reviewer: ```suggestion
    The `QuickSort` and `PartialQuickSort` algorithms are stable since Julia 1.9.
```

`MergeSort` is an O(n log n) stable sorting algorithm but is not in-place – it requires a temporary
array of half the size of the input array – and is typically not quite as fast as `QuickSort`.
It is the default algorithm for non-numeric data.

> The default sorting algorithms are chosen on the basis that they are stable and fast, or
> *appear* to be fast. Usually, `QuickSort` is selected, but `InsertionSort` is preferred for

Reviewer: ```suggestion
The default sorting algorithms are chosen on the basis that they are fast and stable, or
*appear* to be stable. Usually, `QuickSort` is selected, but `InsertionSort` is preferred for
```

small data. You can also explicitly specify your preferred algorithm, e.g.
`sort!(v, alg=PartialQuickSort(10:20))`.

The mechanism by which Julia picks default sorting algorithms is implemented via the
`Base.Sort.defalg` function. It allows a particular algorithm to be registered as the
default in all sorting functions for specific arrays. For example, here is the default
method from [`sort.jl`](https://github.com/JuliaLang/julia/blob/master/base/sort.jl):

```julia
defalg(v::AbstractArray) = DEFAULT_STABLE
```

> You may change the default behavior for specific element type by, e.g., as in
> [InlineStrings.jl](https://github.com/JuliaStrings/InlineStrings.jl/blob/v1.3.2/src/InlineStrings.jl#L903):

Reviewer: ```suggestion
You may change the default behavior for specific types by defining new methods for `defalg`.
For example, [InlineStrings.jl](https://github.com/JuliaStrings/InlineStrings.jl/blob/v1.3.2/src/InlineStrings.jl#L903)
defines the following method:
```

```julia
Base.Sort.defalg(::AbstractArray{<:Union{SmallInlineStrings, Missing}}) = InlineStringSort
```

!!! compat "Julia 1.9"
    The default sorting algorithm (returned by `Base.Sort.defalg`) is guaranteed

>     to be stable since Julia 1.9.

Reviewer: ```suggestion
    to be stable since Julia 1.9. Previous versions had unstable edge cases when sorting numeric arrays.
```
This is the man page, not a docstring so I prefer precision over concision.
